compiler/rustc_hir_typeck/src/pat.rs

```diff
@@ -1941,6 +1966,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         expected: Ty<'tcx>,
         def_bm: BindingMode,
         ti: TopInfo<'tcx>,
+        decl_ctxt: Option<DeclContext>,
```

Comment:
can these three arguments be merged into a shared struct? I think they get mostly passed down together and we're at a point where the argument list is quite hard to read